### PR TITLE
[Runtime] Added TLS compatibility test suite.

### PR DESCRIPTION
**Pull request by alexanderkyte**

### CODE REVIEW

File: mcs/class/System.Net/Test/Tls_Compat/Harness.cs [hunk 1]
```diff
@@ -0,0 +1,252 @@
+using System;
+using System.Threading.Tasks;
+using System.Security.Cryptography.X509Certificates;
+using System.IO;
+using System.Net;
+using System.Net.Sockets;
+using System.Net.Security;
+using System.Collections.Generic;
+
+namespace Macaco
```
Comment (akoeplinger) [inline mcs/class/System.Net/Test/Tls_Compat/Harness.cs]: Does this namespace have a special meaning? :smile: 

Btw. this file doesn't follow the [Mono coding conventions](http://www.mono-project.com/community/contributing/coding-guidelines/).

**Comment by alexanderkyte** [comment]
@akoeplinger: Oops, I had an errant post-commit hook. clang-format's configured for C style, not C# style. And the name had been the name of an internal repo this was copied out of. It's been updated to something more suitable.

**Comment by akoeplinger** [comment]
@alexanderkyte the formatting in Harness.cs still seems wonky, i.e. it doesn't use space before opening parens etc.

**Comment by alexanderkyte** [comment]
Good catch, I forgot that clang-format would change the whitespace.

**Comment by Therzok** [comment]
@alexanderkyte Opening the file in XS and doing Format Document should bring it to Mono standards.

**Comment by migueldeicaza** [comment]
We should not ask our users to fix their system certificates before they can run this test.

Let us ship the CSR that we want to test against directly in Mono.

**Comment by alexanderkyte** [comment]
Like so?

**Comment by migueldeicaza** [comment]
Right, are those integrated now into the test suite?

**Comment by alexanderkyte** [comment]
They're the files that the makefile would call jankca to make. As long as the lock file's there, the test suite will use them. I just ran a sanity check by calling make check from mono/mcs/class/System.Net and it ran just fine for me.

**Comment by migueldeicaza** [comment]
@alexanderkyte It looks like this pull request is now empty.

**Comment by alexanderkyte** [comment]
That's not right.

**Comment by alexanderkyte** [comment]
There we go.

**Pull request closed without merging**